src/apps/DesktopAppPage.vala

```diff
@@ -400,5 +563,470 @@ namespace Ilia {
+                            
+                            hide_app_actions();
+                            
+                            // Print status message to console
+                            stdout.printf("Desktop shortcut created\n");
+                        } catch (Error e) {
+                            stderr.printf("Failed to create desktop shortcut: %s\n", e.message);
+                            hide_app_actions();
+                        }
+                        break;
+                        
+                    case "generic-folder":
+                        try {
+                            // Open the folder containing the desktop file
+                            string folder_path = Path.get_dirname(desktop_file_path);
+                            string[] args = {"xdg-open", folder_path};
+                            Process.spawn_async(null, args, null, SpawnFlags.SEARCH_PATH, null, null);
+                            hide_app_actions();
+                            session_controller.quit();
+                        } catch (Error e) {
+                            stderr.printf("Failed to open app folder: %s\n", e.message);
+                            hide_app_actions();
+                        }
+                        break;
+                        
+                    case "generic-new-workspace":
+                        try {
+                            // Get the app command
+                            var key_file = new KeyFile();
+                            key_file.load_from_file(desktop_file_path, KeyFileFlags.NONE);
+                            string exec_line = key_file.get_string("Desktop Entry", "Exec");
+                            string clean_exec = "";
+                            bool in_arg = false;
+                            for (int i = 0; i < exec_line.length; i++) {
+                                if (exec_line[i] == '%') {
+                                    in_arg = true;
+                                    continue;
+                                }
+                                if (in_arg) {
+                                    in_arg = false;
+                                    continue;
+                                }
+                                clean_exec += exec_line[i].to_string();
+                            }
+                            clean_exec = clean_exec.strip();
+                            string[] args;
+                            var wm_cmd_newws = get_wm_cli(this.session_controller.get_wm_name());
+                            if (wm_cmd_newws != null) {
+                                // For i3/sway: switch to next workspace and launch app
+                                string launch_cmd = wm_cmd_newws + " workspace next && " + clean_exec;
+                                args = {"sh", "-c", launch_cmd};
+                            } else {
+                                args = {"sh", "-c", clean_exec};
```

Comment:
Use `systemd-run` to launch applications. You can refer to `get_runner_app_info()` function on how to do that. See if you can use it directly to get the execute command.